File: notas.m
clc; 
clear all; 
close all;

retry= true;
while retry
  try
    fprintf("Elija una opcion: (1.Promedio y estado del alumno,   2.Historial)");
    i=(input ("\n Opcion a elegir: "));
    if i>3||i<0
      fprintf("\n Numero erroneo, vuelva a ingresar: ");
    endif
    if i==1
      n1=(input('Ingrese la primer nota: '));
      n2=(input('Ingrese la segunda nota: '));
      n3=(input('Ingrese la tercera nota: '));
      prom=((n1+n2+n3)/3);
      if 60<prom||60==prom
          fprintf('Alumno Aprobado \n');
          estado='Aprobado';
          fprintf('El promedio de notas es: %d \n', prom);
      endif
      if 60>prom
          fprintf('Alumno Reprobado \n');
          estado='Reprobado';
          fprintf('El promedio de notas es: %d \n', prom);
      endif
      %Conectar con PGADMIN
      conn = pq_connect(setdbopts('dbname','tarea1','host','localhost',
      'port','5432','user','postgres','password','jblue5queso'));
      params=cell(1,5);
      params{1,1}=n1;
      params{1,2}=n2;
      params{1,3}=n3;
      params{1,4}=prom;
      params{1,5}=estado;
      N=pq_exec_params(conn, "insert into notas(n1, n2, n3, promedio, estado) values($1,$2,$3,$4,$5);",params); %insertar datos en la tabla
      %Texto
      save('notas.txt','n1', 'n2', 'n3', 'prom','estado','-ascii');

    endif
    if i==2
       conn = pq_connect(setdbopts('dbname','tarea1','host','localhost',
       'port','5432','user','postgres','password','jblue5queso'));
       N=pq_exec_params(conn, "Select * from notas;");%historial de la tabla
       disp(N);
       retry=false;
    endif
 
  catch
      fprintf("Solo se permiten valores numericos, vuelava a selecionar %d \n");
      msg = lasterror.message;
      fprintf(msg);
    end_try_catch
 endwhile

File: notas.py
import psycopg2
from tabulate import tabulate

def  post(x,y,z, w, v):
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "jblue5queso",
            dbname = "tarea1"
            )
    except:
        print("Sin Conexion Exitosa\n")
            
    cursor = conexion.cursor()
    cursor.execute("INSERT INTO notas(n1, n2, n3, promedio, estado) VALUES (%s, %s, %s, %s, %s);", (x, y, z, w, v))
    conexion.commit()
    cursor.close()
    conexion.close()

def ret():
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "jblue5queso",
            dbname = "tarea1"
            )
    except:
        print("Sin Conexion Exitosa\n")
    
    cursor = conexion.cursor()
    cursor.execute("SELECT * from notas;")
    print(tabulate(cursor, headers=["ide", "n1", "n2", "n3", "promedio", "estado"], tablefmt="orgtbl", numalign ="center"))

while True:
    try:
        print("Elija una opcion: (1. Promedio y estado del alumno,   2.Historial,   3.Salir)")
        x=int(input("Elija la opcion:\n"))
        if x>3 or x<=0:
            print("Valor ingresado fuera del rango del menu.\n")
        elif x==1:
            estado=""
            n1=int(input("Ingrese la primer nota: "))
            n2=int(input("Ingrese la segunda nota: "))
            n3=int(input("Ingrese la tercera nota: "))
            prom=((n1+n2+n3)/3)
            if 60<prom or 60==prom:
                print("Alumno Aprobado")
                estado="Aprobado"
                print("El promedio de notas es: " + str(prom))
            
            if 60>prom:
                print("Alumno Reprobado")
                estado="Reprobado"
                print("El promedio de notas es: " + str(prom))
            post(n1, n2, n3, prom, estado)
        elif x==2:
            ret()
        elif x==3:
            break
    except Exception as e:
        print("\n A ingresado un valor o Caracter no valido por favor, ingrese una opcion correcta:\n")
        print(repr(e))

exit(0)